test/unit/transaction-runner-test.coffee

```diff
@@ -1452,85 +1454,93 @@ describe 'TransactionRunner', ->
 
-          runner.executeAllTransactions [], runner.hooks, () ->
+          runner.executeAllTransactions [], runner.hooks, (err) ->
             call = configuration.emitter.emit.getCall(0)
             assert.ok configuration.emitter.emit.calledWith "test error"
             assert.include call.args[1].message, 'require'
-            done()
+            done(err)
 
         it 'should not have aceess to current context', (done) ->
           contextVar = "this"
           functionString = """
-          function(transaction){
+          function(transactions){
             contextVar = "that";
           }
           """
           runner.hooks.beforeAll functionString
 
-          runner.executeAllTransactions [], runner.hooks, () ->
+          runner.executeAllTransactions [], runner.hooks, (err) ->
             assert.equal contextVar, 'this'
-            done()
+            done(err)
 
         it 'should have access to the hook stash', (done) ->
           functionString = """
-          function(transaction){
-            stash['prop'] = 'that';
+          function(transactions){
+            stash.prop = 'that';
           }
           """
           runner.hooks.beforeAll functionString
 
-          runner.executeAllTransactions [], runner.hooks, () ->
+          runner.executeAllTransactions [], runner.hooks, (err) ->
             assert.notOk configuration.emitter.emit.calledWith "test error"
-            done()
+            done(err)
 
         it 'should be able to modify hook stash', (done) ->
           functionString = """
-          function(transaction){
-            stash['prop'] = 'that';
+          function(transactions){
+            stash.prop = 'that';
           }
           """
           runner.hooks.beforeAll functionString
 
-          runner.executeAllTransactions [], runner.hooks, () ->
+          runner.executeAllTransactions [], runner.hooks, (err) ->
             assert.notOk configuration.emitter.emit.calledWith "test error"
             assert.property runner.hookStash, 'prop'
-            done()
+            done(err)
 
         it 'should be able to modify transactions', (done) ->
           functionString = """
           function(transactions){
-            transactions['prop'] = 'that';
+            transactions[0].name = 'Changed!';
           }
           """
           runner.hooks.beforeAll functionString
 
-          transactions = {'some': 'mess'}
+          transactions = [
+            name: 'Test!',
+            request: {headers: {}, body: ''},
+            skip: true,
+          ]
 
-          runner.executeAllTransactions transactions, runner.hooks, () ->
+          runner.executeAllTransactions transactions, runner.hooks, (err) ->
             call = configuration.emitter.emit.getCall(0)
             assert.notOk configuration.emitter.emit.calledWith "test error"
-            assert.property transactions, 'prop'
-            done()
+            assert.equal transactions[0].name, 'Changed!'
```

Comment:
Trust me! The previous test used dummy structure to test something. Since async.js got more strict about what can go inside, the dummy structure didn't work anymore. I replaced it with more real-world examples to make the test work.